six/modules/c++/cphd/source/PVPBlock.cpp

```diff
@@ -288,32 +288,31 @@ void PVPBlock::PVPSet::read(const Pvp& p, sys::ubyte* dest) const
 /*
  * Initialize PVP Array with a data object
  */
-PVPBlock::PVPBlock(const Data& d, const Pvp& p)
+PVPBlock::PVPBlock(const Pvp& p, Data& d)
 {
     mPvp = p;
-    mNumBytesPerVector = mPvp.getReqSetSize()*sizeof(double);
+    mNumBytesPerVector = d.getNumBytesPVPSet();
     mData.resize(d.getNumChannels());
     for (size_t ii = 0; ii < d.getNumChannels(); ++ii)
     {
         mData[ii].resize(d.getNumVectors(ii));
     }
-    size_t calculateBytesPerVector = getNumBytesPVPSet();
+    size_t calculateBytesPerVector = mPvp.getReqSetSize()*sizeof(double);
     if (six::Init::isUndefined<size_t>(mNumBytesPerVector) ||
         calculateBytesPerVector > mNumBytesPerVector)
     {
         mNumBytesPerVector = calculateBytesPerVector;
+        d.numBytesPVP = calculateBytesPerVector;
```

Comment:
I'm updating the numBytesPVP metadata here instead of the CPHDWriter::writeMetadata.
I thinkthis makes the most sense, because the PVPBlock constructor requires a filled pvp metadata object specifying all the parameters that will exist in the PVP block. 
If the pvp metadata is changed, a new PVPBlock object would have to be created, so the numBytesPVP metadata would just get updated everytime a new PVPBlock is created.

Reply:
So under what conditions would the condition `calculateBytesPerVector > mNumBytesPerVector` on line 302 be true? Having the  bytesPerVector value go from `Data` to `PVPBlock` and also from `PVPBlock` to `Data` in the same function is hard to follow. Would it make sense to just throw if the `Data` object doesn't match the `Pvp`?

Reply:
`calculateBytesPerVector > mNumBytesPerVector` would be true if the size of the vector parameter set is greater than the metadata specified by the user? (I.e. the metadata for PVP and the numPVPBytes metadata are inconsistent)
I guess this would be a case where the onus is on the user to make sure the metadata is consistent. So I could just throw an error. (That's what you're suggesting right?)

Reply:
Right